### ISSUE TITLE
Item click-through

### ISSUE THREAD

**Issue by SlimeDog** [RockinChaos/ItemJoin#225]
Spigot 1.14.4 latest or Spigot 1.13.2 final
ItemJoin 5.0.4-b285
DeluxeMenus 1.12.0
Skript 2.4-beta5

Right-click on CLOCK item opens the DeluxeMenus menu, but also acts on the iron_door at which the click is aimed, which triggers a Skript to toggle the iron_door. This occurs even though _itemflags_ includes _cancel-events_. The issue does not occur if the right-click would trigger a vanilla toggle event (for example, if the target were a wooden door).

Similar to, or perhaps reappearance of, #163.

ItemJoin items.yml (relevant portion):
```
  changeWorld-tool:
    id: CLOCK
    slot: 8
    count: 1
    name: '&bNavigate'
    lore:
      - 'Teleport elsewhere'
    commands:
      right-click:
        - 'deluxemenus open navigation'
    enchantment: FIRE_ASPECT:1
    itemflags: cancel-events, death-drops, hide-attributes, hide-durability, inventory-modify, placement, self-drops, unbreakable
    triggers: join, respawn, world-change
    permission-node: 'deluxemenus.open'
    enabled-worlds: world, world_lobby, world_blackdog
```
and config.yml (sans comments):
```
config-Version: 7

Language: 'English'

General:
  CheckforUpdates: false
  Metrics-Logging: false
  Log-Coloration: false
  Log-Commands: true
  Debugging: false

Settings:
  HeldItem-Slot: 0
  DataTags: true

Permissions:
  Obtain-Items: false
  Obtain-Items-OP: false
  Commands-Get: false
  Commands-OP: false
  Movement-Bypass: false

Clear-Items:
  Type: ITEMJOIN
  Delay-Tick: 2
  Join: DISABLED
  World-Switch: DISABLED
  Options: PROTECT
  Bypass: DISABLED

Active-Commands:
  commands:
    - 'itemjoin get <item> <player>'
    - 'first-join: say This is a command only executed once per world, per player.'
  triggers: JOIN
  enabled-worlds: DISABLED

Prevent:
  Pickups: DISABLED
  itemMovement: DISABLED
  Bypass: DISABLED

softDepend:
  Multiverse-Core: true
  Multiverse-Inventories: true
  PlaceholderAPI: true
  Vault: true
  WorldGuard: true

  AuthMe: false
  BetterNick: false
  HeadDatabase: false
  MyWorlds: false
  PerWorldInventory: false
  PerWorldPlugins: false
  TokenEnchant: false
  xInventories: false
```
and the Skript:
```
on rightclick on iron door:
	toggle the block
	cancel the event
```

**Comment by RockinChaos** [comment]
Is this specific to the latest dev build (occurring after you updated) or do you think this has been an existing issue prior to build #285?

**Comment by SlimeDog** [comment]
I agree. The event in question is not vanilla. I will confirm when I get back to my desktop.

**Comment by SlimeDog** [comment]
Were you able to replicate?

**Comment by RockinChaos** [comment]
Sorry I was just about to complete this issue today/tomorrow. I have been working on it and I have replicated the issue. I am still working on a method to resolve this though.

I have been spending a majority of the time finishing the item creator gui which you will see pushed to the snapshot branch later tomorrow.

Expect to see a reply on this issue tomorrow or the day after.

**Comment by SlimeDog** [comment]
Sounds good. Thanks.

**Comment by SlimeDog** [comment]
Yes! This works as expected for all tested variations (see bullet points in "Analysis" above) with build 289.

**Comment by SlimeDog** [comment]
Feature request: It would be nice if the version string in plugins.yml included the build number. For example, instead of:
- 5.0.4-SNAPSHOT

it would be one of:
- 5.0.4-SNAPSHOT-b289
- 5.0.4-b289

**Comment by RockinChaos** [comment]
This is something I actually intensely worked on when I first started the Jenkins repo site. However (At the time) it wasn't really feasible and was quite complex as Jenkins wasn't really friendly with modifying yaml files during compilation.

However, I will take another look at it and see what has changed. A little birdy told me it has gotten a lot easier now, and honestly would be VERY useful for me to keep track of things.

**Comment by RockinChaos** [comment]
Done :)

http://ci.craftationgaming.com/job/ItemJoin/292/

-- I have no idea how this will affect the plugin checking for updates, I haven't tested it yet lol.
Build number exists on the end of the jar file as well as the plugin.yml so typing /itemjoin (will display the build number)

**Comment by SlimeDog** [comment]
Nice. My automagical update-checker displays:
```
plugin                        source    jar version          running version      latest release       date         status
----------------------------- --------- -------------------- -------------------- -------------------- ------------ --------
ItemJoin                      Jenkins   5.0.4-b289           5.0.4-b289 *         5.0.4-b290           2019-09-13   update
```

**Comment by SlimeDog** [comment]
however...
```
[22:06:50] [Server thread/INFO]: [ItemJoin] Enabling ItemJoin v5.0.4-SNAPSHOT-b292
[22:06:50] [Server thread/INFO]: [ItemJoin] Fetched [{ Multiverse-Core, Multiverse-Inventories, WorldGuard, PlaceholderAPI, Vault }]^[[m
[22:06:50] [Server thread/ERROR]: Error occurred while enabling ItemJoin v5.0.4-SNAPSHOT-b292 (Is it up to date?)
java.lang.NumberFormatException: For input string: "504-292"
        at jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054) ~[?:?]
        at jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110) ~[?:?]
        at java.lang.Double.parseDouble(Double.java:543) ~[?:?]
        at me.RockinChaos.itemjoin.handlers.UpdateHandler.<init>(UpdateHandler.java:29) ~[?:?]
        at me.RockinChaos.itemjoin.handlers.ConfigHandler.generateData(ConfigHandler.java:47) ~[?:?]
        at me.RockinChaos.itemjoin.ItemJoin.onEnable(ItemJoin.java:14) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:352) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:417) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugin(CraftServer.java:461) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugins(CraftServer.java:375) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:449) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at net.minecraft.server.v1_14_R1.DedicatedServer.init(DedicatedServer.java:266) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:775) [spigot-1.14.4.jar-2019-09-13-0714:git-Spigot-94af569-6ddeb98]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by RockinChaos** [comment]
Sorry about that, had a feeling that would happen.
Did some tweaking and (I did test it) and it seems to be working, as well as it seems happy with the existing release builds on spigotmc so this seems like a fantastic edition.
http://ci.craftationgaming.com/job/ItemJoin/293/

**Comment by RockinChaos** [comment]
This bug has been officially fixed in the release of ItemJoin v5.0.4.
You can grab it here; https://www.spigotmc.org/resources/itemjoin.12661/download?version=295178

Please read the changelog as the documentation has not been updated yet;
https://github.com/RockinChaos/ItemJoin/wiki/Recent-Changes
